src/txn/txn_timestamp.c

```diff
@@ -688,8 +688,8 @@ __wt_txn_set_commit_timestamp(
 
 	if (!F_ISSET(txn, WT_TXN_HAS_TS_PREPARE)) {
 		/*
-		 * For a non-prepared transactions the commit timestamp should
-		 * not be less than the stable timestamp.
+		 * For non-prepared transactions the commit timestamp should
+		 * not be less than the oldest timestamp.
 		 */
 		if (has_oldest_ts && commit_ts < oldest_ts)
```

Comment:
the condition should be `commit_ts <= oldest_ts`

Reply:
As discussed we are going to keep it as just less than the oldest timestamp as the behavior would conflict with the rounding up to oldest timestamp logic.